pkg/endpoint/endpoint.go

```diff
@@ -257,3 +265,16 @@ func networkAndAddress(str string) (string, string) {
 	}
 	return "", parts[0]
 }
+
+// validateStorageEndpoint validates the given string can be used to infer backend and optionally extract connection
+// info
+func validateStorageEndpoint(str string) error {
+	if strings.Contains(str, "://") {
+		return nil
+	}
+	if str == "" || str == "sqlite" {
+		return nil
+	}
+	// without this check kine will silently run embedded sqlite, despite user intending to use a passed backend
+	return fmt.Errorf("invalid storage endpoint [%s]. Non-empty storage endpoint should be of the format <network>://<address>", str)
```

Comment:
1. Don't log the DSN, as it may contain db credentials that we should not log
2. Lets call this the datastore endpoint in the message, instead of the storage endpoint, as that is the name of the CLI flag in K3s and RKE2.
3. Use standard component names as per https://datatracker.ietf.org/doc/html/rfc3986#section-3
```suggestion
	return errors.New("invalid datastore endpoint; endpoint should be a DSN URI in the format <scheme>://<authority>")
```

Reply:
@brandond if I move the validation to the switch under the network == "" case, I don't think this message makes sense. If network == "" then the string is clearly not of that format but we reserve two cases where it is acceptable.